src/SFML/Graphics/Texture.cpp

```diff
@@ -120,6 +121,49 @@ Texture::~Texture()
     }
 }
 
+////////////////////////////////////////////////////////////
+Texture::Texture(Texture&& right) noexcept :
+m_size(std::exchange(right.m_size, {0U, 0U})),
+m_actualSize(std::exchange(right.m_actualSize, {0U, 0U})),
+m_texture(std::exchange(right.m_texture, 0)),
+m_isSmooth(std::exchange(right.m_isSmooth, false)),
+m_sRgb(std::exchange(right.m_sRgb, false)),
+m_isRepeated(std::exchange(right.m_isRepeated, false)),
+m_fboAttachment(std::exchange(right.m_fboAttachment, false)),
+m_cacheId(std::exchange(right.m_cacheId, 0))
```

Comment:
Is it ok to reuse cache id? Asking because `Texture::swap` doesn't reuse cache ids.

Reply:
I'm going to wait for other's input.

I thought it would be "proper" because it's being moved.

Reply:
The cache ID is used (primarily?) in two places:

https://github.com/SFML/SFML/blob/3745ea23367d275d1d25105c81c3bcd243e00a53/src/SFML/Graphics/RenderTarget.cpp#L709-L711

https://github.com/SFML/SFML/blob/3745ea23367d275d1d25105c81c3bcd243e00a53/src/SFML/Graphics/Text.cpp#L399-L401

If we move a texture, then the texture itself didn't/shouldn't have changed, as such, I'd guess, that the ID should remain the same. This would certainly be an interesting thing to test. Plus also in general to check if the texture set on a sprite remains usable after a move.

Reply:
I did test it with the example code above and it worked as I expected. 

Admittedly I didn't test with moving the texture back and then drawing.

Reply:
We had the discussion over at https://github.com/SFML/SFML/pull/2479#discussion_r1152221932 

Lets go with re-using for now and if people start reporting issue, we can try to analyze more closely.